controllers/order.js

```diff
@@ -1,114 +1,59 @@
-        "time": "2022-04-11T12:20:05",
-        "items": [
-            {
-                "id": "item001",
-                "name": "蛋包飯",
-                "price": 80,
-                "quantity": 1,
-                "note": "飯不要太多",
-                "status": "RAW"
-            },
-            {
-                "id": "item010",
-                "name": "珍珠鮮奶茶",
-                "price": 60,
-                "quantity": 1,
-                "note": null,
-                "status": "RAW"
-            },
-            {
-                "id": "item007",
-                "name": "泡芙",
-                "price": 25,
-                "quantity": 1,
-                "note": null,
-                "status": "RAW"
-            }
-        ]
-    },
-    {
-        "id": "order002",
-        "tableNo": "12D",
-        "totalPrice": 765,
-        "time": "2022-04-11T13:15:12",
-        "items": [
-            {
-                "id": "item002",
-                "name": "義大利麵",
-                "price": 100,
-                "quantity": 4,
-                "note": "兩個青醬兩個紅醬",
-                "status": "RAW"
-            },
-            {
-                "id": "item004",
-                "name": "水果茶",
-                "price": 90,
-                "quantity": 2,
-                "note": null,
-                "status": "RAW"
-            },
-            {
-                "id": "item009",
-                "name": "薯條",
-                "price": 40,
-                "quantity": 2,
-                "note": "要番茄醬",
-                "status": "RAW"
-            },
-            {
-                "id": "item008",
-                "name": "洋蔥圈",
-                "price": 45,
-                "quantity": 1,
-                "note": null,
-                "status": "RAW"
-            },
-            {
-                "id": "item010",
-                "name": "珍珠鮮奶茶",
-                "price": 60,
-                "quantity": 1,
-                "note": null,
-                "status": "RAW"
-            }
-        ]
-    },
-    {
-        "id": "order003",
-        "tableNo": 3,
-        "totalPrice": 120,
-        "time": "2022-04-11T14:01:18",
-        "items": [
-            {
-                "id": "item006",
-                "name": "披薩",
-                "price": 120,
-                "quantity": 1,
-                "note": null,
-                "status": "RAW"
+
+const postOrder = async (req, res)=>{
+    let order = req.body
+    // todo is there a way to make these query a transaction?
+    try{
+        await dbMutation(`INSERT INTO \`Order\` VALUES('${order.id}', '${order.tableNo}', ${order.totalPrice}, '${order.time}' )`)
+        order.items.forEach(async (item)=>{
+            let Order_Item_InfoId = order.id+"_"+item.id
+            await dbMutation(`INSERT INTO \`Order_Item_Info\` VALUES('${Order_Item_InfoId}', '${item.quantity}', '${item.note}')`)
+    
+            let Order_ItemId = "Order_Item" + Math.floor(Math.random()*1000)
```

Comment:
actually the best practice is to use a **UUID** https://www.npmjs.com/package/uuid for generating an **id**
an example given there [uuid.v4()](https://www.npmjs.com/package/uuid#uuidv4options-buffer-offset)

Reply:
I know that. But if I use uuid.v4(), the id format will be different from the sample data the frontend guys offered.